src/main/java/org/onedatashare/server/module/http/HttpResource.java

```diff
@@ -215,40 +216,35 @@ private static String getName(String uri) {
     }
 
     /**
-     * Class to parse the size parameter in the webpage
+     * parses the Size String from the webpage
      */
-    private static class SizeParser {
-
-        public static long getBytes(String sizeString) {
-            int sizeStringLength = sizeString.length();
-            char multiplier = sizeString.charAt(sizeStringLength - 1);
-            long multiply = 1;
-            switch (multiplier) {
-                case 'T':
-                    multiply <<= 40;
-                case 'G':
-                    multiply <<= 30;
-                    break;
-                case 'M':
-                    multiply <<= 20;
-                    break;
-                case 'K':
-                    multiply <<= 10;
-            }
+    public static long getBytes(String sizeString) {
+        int sizeStringLength = sizeString.length();
+        char multiplier = sizeString.charAt(sizeStringLength - 1);
+        long multiply = 1;
+        switch (Character.toUpperCase(multiplier)) {
+            case TERA:
+                multiply <<= 10;
+            case GIGA:
+                multiply <<= 10;
```

Comment:
Are the numbers correct?

Reply:
Yes. Because there are no break statements.

Reply:
Makes sense. Did not catch that.
Although it looks fancy, it can easily confuse any reader. It would be better practice to put break statements with standard conversion values. 
Or retain it and put some comment above it.

Reply:
Also, if the switch enters TERA, it would require 4 computations to get the size. With break it and standard value, it will get done in one computation.

Reply:
Yes. It'll be. Made the change.